sdk-cpp/src/do_errors.cpp

```diff
@@ -1,26 +1,38 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-#if defined(DO_ENABLE_EXCEPTIONS)
-
 #include "do_errors.h"
 
 namespace microsoft
 {
 namespace deliveryoptimization
 {
 
-const error_category& error_category_instance()
+const error_category& do_category()
 {
     static error_category instance;
     return instance;
 }
 
+std::error_code make_error_code(std::errc e)
+{
+    return std::error_code(DO_ERROR_FROM_STD_ERROR(e), do_category());
```

Comment:
>DO_ERROR_FROM_STD_ERROR [](http://example.com/codeflow?start=27&length=23)

Need to fix tests to avoid doing this but it's okay for now.

Reply:
I'll sign off.

Reply:
Hmm I'm not sure I follow - I think we do need to do this in order to make use of the same error code macros/convention. the std::errc enum are a set of positive integer values, and so need to call DO_ERROR_FROM_STD_ERROR to make them hresult-like.